modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/RoleResourceImpl.java

```diff
@@ -70,6 +71,17 @@ public void deleteOrganizationRoleUserAccountAssociation(
 			userAccountId, organization.getGroupId(), new long[] {roleId});
 	}
 
+	public void deleteRoleByExternalReferenceCodeUserAccountAssociation(
+			String roleExternalReferenceCode, Long userAccountId)
+		throws PortalException {
+
+		com.liferay.portal.kernel.model.Role serviceBuilderRole =
+			_getServiceBuilderRole(roleExternalReferenceCode);
+
+		_userService.deleteRoleUser(
+			serviceBuilderRole.getRoleId(), userAccountId);
```

Comment:
Why here are you calling the _userService instead of call the same method but by id? It's not wrong but in the next one `getRoleByExternalReferenceCode` you are doing differently. You should try to be consistent as much as possible